### PR TITLE
ISSUE-16 - Construct cargo_geiger PackageId from a cargo_metadata

### PR DESCRIPTION
**Pull request by jmcconnell26**
PackageId:
* Add method to construct Source from cargo metadata
* Add tests Source construction
* Remove further usages of package set

Signed-off-by: joshmc <josh-mcc@tiscali.co.uk>

### CODE REVIEW

File: cargo-geiger/src/mapping/geiger.rs [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+fn handle_source_repr(source_repr: &str) -> cargo_geiger_serde::Source {
+    let mut source_repr_vec = source_repr.split('+').collect::<Vec<&str>>();
+
+    let source_type = source_repr_vec[0];
+
+    match source_type {
+        "registry" => {
+            cargo_geiger_serde::Source::Registry {
+                // It looks like cargo metadata drops this information
+                name: String::from("crates.io"),
```
Comment (jmcconnell26) [inline cargo-geiger/src/mapping/geiger.rs:32]: I'm unsure on this, it looks like it gets dropped by cargo_metadata, I'm not sure if we want to parse this from somewhere else (or even where we would parse it from)?

**Comment by jmcconnell26** [comment]
This should now be ready to go, sorry about the delay, I had a wee bit of trouble with the paths in Windows.

File: cargo-geiger/src/mapping/geiger.rs (hunk 1 above)
Comment (anderejd) [reply cargo-geiger/src/mapping/geiger.rs:31]: I think the comment is good enough for now.

**Comment by anderejd** [comment]
> This should now be ready to go, sorry about the delay, I had a wee bit of trouble with the paths in Windows.

No problem, I'm taking my sweet time getting to reviewing as well it seems. Great work!

**Pull request merged**